src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -2352,7 +2352,7 @@ def forward(
         spatial_shapes = torch.as_tensor(spatial_shapes, dtype=torch.long, device=source_flatten.device)
         level_start_index = torch.cat((spatial_shapes.new_zeros((1,)), spatial_shapes.prod(1).cumsum(0)[:-1]))
         valid_ratios = torch.stack([self.get_valid_ratio(m) for m in masks], 1)
-        valid_ratios = valid_ratios.float()
+        valid_ratios = valid_ratios.to(pixel_values.dtype)
```

Comment:
`reference_points_input` is created by `reference_points` and `valid_ratios`

https://github.com/huggingface/transformers/blob/f3c8b18053054674b9fe8e2bd4e555e52b1ab857/src/transformers/models/grounding_dino/modeling_grounding_dino.py?plain=1#L1911